code/obj/machinery.dm

```diff
@@ -368,6 +368,6 @@
 	var/area/A1 = get_area(src)
 	. = ..()
 	var/area/A2 = get_area(src)
-	if(A1 != A2)
+	if(!isnull(A1?.machines) && !isnull(A2?.machines) && A1 != A2)
```

Comment:
fixing the symptom not the problem, .machines will never be null, by looking at references to it
However, it is valid to check if A1 and A2 exist / not null, which would be the proper fix